AgriParcel/schema.json

```diff
@@ -1,162 +1,180 @@
+            },
+            "hasAgriParcelChildren": {
+                "type": "array",
+                "description": "Relationship. Related sub AgriParcel records to which this entity relates",
+                "items": {"$ref": "https://smart-data-models.github.io/data-models/common-schema.json#/definitions/EntityIdentifierType"}
+            },
+            "hasAgriCrop": {
+                "anyOf": [
+                    {
+                        "type": "string",
+                        "minLength": 1,
+                        "maxLength": 256,
+                        "pattern": "^[\\w\\-\\.\\{\\}\\$\\+\\*\\[\\]`|~^@!,:\\\\]+$",
+                        "description": "Property. Identifier format of any NGSI entity"
+                    },
+                    {
+                        "type": "string",
+                        "format": "uri",
+                        "description": "Property. Identifier format of any NGSI entity"
+                    }
+                ],
+                "description": "Relationship. Reference to the crop associated with this parcel"
+            },
+            "hasAgriSoil": {
+                "anyOf": [
+                    {
+                        "type": "string",
+                        "minLength": 1,
+                        "maxLength": 256,
+                        "pattern": "^[\\w\\-\\.\\{\\}\\$\\+\\*\\[\\]`|~^@!,:\\\\]+$",
+                        "description": "Property. Identifier format of any NGSI entity"
+                    },
+                    {
+                        "type": "string",
+                        "format": "uri",
+                        "description": "Property. Identifier format of any NGSI entity"
+                    }
+                ],
+                "description": "Relationship. Reference to the soil associated with this parcel of land"
+            },
+            "cropStatus": {
+                "type": "string",
+                "description": "Property. Enum:'seeded, justBorn, growing, maturing, readyForHarvesting'. A choice from an enumerated list describing the crop planting status",
+                "enum": [
+                    "seeded",
+                    "justBorn",
+                    "growing",
+                    "maturing",
+                    "readyForHarvesting"
+                ]
+            },
+            "lastPlantedAt": {
+                "type": "string",
+                "format": "date-time",
+                "description": "Property. Indicates the date when the crop was last planted"
+            },
+            "hasDevices": {
+                "type": "array",
+                "description": "Property. Reference to the IoT devices associated with this parcel i.e. sensors, controls.",
+                "items": {"$ref": "https://smart-data-models.github.io/data-models/common-schema.json#/definitions/EntityIdentifierType"}
+            },
+            "soilTextureType": {
```

Comment:
I guess that an array with enumeration will do this much simpler and more extensible